akka-typed/src/main/scala/akka/typed/cluster/ddata/scaladsl/Replicator.scala

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.cluster.ddata.scaladsl
+
+import akka.actor.NoSerializationVerificationNeeded
+
+import akka.cluster.{ ddata ⇒ dd }
+import akka.cluster.ddata.Key
+import akka.cluster.ddata.ReplicatedData
+import akka.typed.ActorRef
+import akka.typed.Behavior
+import akka.typed.cluster.ddata.internal.ReplicatorBehavior
+
+object Replicator {
+
+  def behavior(settings: ReplicatorSettings): Behavior[Command[_]] =
+    ReplicatorBehavior.behavior(settings)
```

Comment:
A bit vague but: This compared to the much more locked down API versions for Singleton/Sharding, does it make more sense to give users access to running a replicator wherever they want in their tree? (and counter - should the Singleton/Sharding ones be more like this returning a behaviour and leave more options for the user?)

Reply:
There will also be an extension as a convenience, same as untyped ddata. It must still be possible to start your own Replicator instances because it should be possible to run several with different settings (e.g. performance reasons).

Comparing to singleton. In the new api `spawn` is starting both manager and proxy so that makes sense. We could consider if `spawnManager` and `spawnProxy` should instead only be `Behavior` factories and user performs the `ctx.spawn`. More flexible and we said that those are for power user api anyway. Drawback might be that it's easier to get actor paths wrong.

Sharding must be an extension.

Reply:
I think that is good, power-user returns behaviour and it's up to user to get it right, I'll change the singleton API

Reply:
ok